app/controllers/orders_controller.rb

```diff
@@ -27,10 +29,20 @@ def update
     order.driver_id = @current_user.id
     order.state = 'in progress'
     order.save
-    render json: { 'current_order' => order.id }
+    render json: { 'current_order' => order }
     send_email_to_client(order.id, order.client_id)
   end
 
+  def cancel
+    if @current_user.try(:instance_of?, Driver)
+      return render json: { 'error' => 'You are not allowed' }, status: 422
```

Comment:
@domenic99 this is too harsh :laughing: 
It's better to respond with 'You are not allowed to cancel an order' and status 403 :wink: